source/val/validation_state.h

```diff
@@ -859,6 +859,11 @@ class ValidationState_t {
            qcom_image_processing_consumers_.end();
   }
 
+  // Get the list of line lengths for a given result ID of a DebugSource
+  // instruction Will create a new vector if DebugSource is not found
+  std::vector<uint32_t>& GetDebugSourceLineLength(uint32_t id,
+                                                  bool use_last_id);
```

Comment:
I feel like it would be safer to not provide the `use_last_id` param and have the builder get the correct id instead to access the vector.